server/eventlog.go

```diff
@@ -425,8 +430,14 @@ func getGrubState(hash crypto.Hash, events []*pb.Event) (*pb.GrubState, error) {
 			continue
 		}
 
-		if event.GetUntrustedType() != IPL {
-			return nil, fmt.Errorf("invalid event type for PCR%d, expected EV_IPL", index)
+		// Check if the event type is EV_IPL or a valid EV_EVENT_TAG.
+		if event.GetUntrustedType() != IPL && !isValidEventTag(event) {
```

Comment:
This logic is very complex. I don't think we need to check the validity of the event. Not all events have digests that match the event data. See the "GrubFile" parsing. Really, we should check these events are all pre-ExitBootServices if we want to be sure all of these events came from a BootLoader. There is no *guarantee* it's from GRUB even then. For this, we rely on SecureBoot or checking the measurements in EfiState. See getEfiState.

Reply:
Done.
Instead of checking the validity of the EV_EVENT_TAG, we'll just skip this type of event.